core/src/main/scala/org/apache/spark/sql/delta/commands/MergeIntoCommand.scala

```diff
@@ -544,8 +544,9 @@ case class MergeIntoCommand(
         // In some cases (e.g. insert-only when all rows are matched, insert-only with an empty
         // source, insert-only with an unsatisfied condition) we can write out an empty insertDf.
         // This is hard to catch before the write without collecting the DF ahead of time. Instead,
-        // we can just accept only the AddFiles that actually add rows.
-        case a: AddFile => a.getNumLogicalRecords > 0
+        // we can just accept only the AddFiles that actually add rows or
+        // when we don't know the number of records
+        case a: AddFile => a.numLogicalRecords.getOrElse(1L) > 0
```

Comment:
nit: `a.numLogicalRecords.forall(_ > 0)`.

Reply:
Thanks. I was searching for a more elegant way to do this